### PR TITLE
[guilib] fix hint text vanish on unfocus - closes #15765

### LINKED ISSUE #15765: [games] In most games only one controller works
### Describe the bug
Here is a clear and concise description of what the problem is:

I use kodi on an odroid c2. Attached to the odroid are two buffalo classic USB gamepads. Both of them are working fine inside of Kodi. When playing 
multiplayer games, however, only one of them seems to work e.g. the option for two players is grayed out or simply not selectable. This behaviour however is not consistent. I have at least one game where it is working, however most are 
not.

## Expected Behavior

Both controllers should work in games to play multiplayer games. 

### To Reproduce

Install a fresh kodi 18.1 using coreelec on odroid c2. Add two buffalo USB controllers to the odroid. Configure the controllers in system > controllers to be sure. Confirm that both controllers work fine in kodi. Add a multiplayer game (or play from IAGL) and try to play it with two controllers. 

### Debuglog

https://gist.github.com/cpfaff/b18294a2aa21a576877140f0571f4629

### Your Environment

 - [ ] Android
 - [ ] iOS
 - [x] Linux
 - [ ] OSX
 - [ ] Raspberry-Pi
 - [ ] Windows
 - [ ] Windows UWP

 - Operating system version/name: Coreelec 9.0.1
 - Kodi version: 18.1

*note: Once the issue is made we require you to update it with new information or Kodi versions should that be required.
Team Kodi will consider your problem report however, we will not make any promises the problem will be solved.*

### PR DESCRIPTION
**Pull request by mkortstiege**
@Montellese, @xhaggi for review please.

### CODE REVIEW

File: xbmc/guilib/GUIEditControl.cpp [hunk 1]
```diff
@@ -743,7 +743,7 @@ void CGUIEditControl::ValidateInput()
 
 void CGUIEditControl::SetFocus(bool focus)
 {
-  m_smsTimer.Stop();
+  UpdateText();
```
Comment (xhaggi) [inline xbmc/guilib/GUIEditControl.cpp]: there is no need to send an update (validate input +  send click message etc.) in `UpdateText()`. You should use `UpdateText(false)` which only invalidate the control and stops m_smsTimer.

**Comment by xhaggi** [comment]
@mkortstiege might be back-ported as it's easy to do ;)

**Comment by mkortstiege** [comment]
@xhaggi thanks, updated. Since `SetInvalid()` is enough i've removed the call to `UpdateText`.

File: xbmc/guilib/GUIEditControl.cpp [hunk 2]
```diff
@@ -746,4 +746,5 @@ void CGUIEditControl::SetFocus(bool focus)
   m_smsTimer.Stop();
   g_Windowing.EnableTextInput(focus);
   CGUIControl::SetFocus(focus);
+  SetInvalid();
```
Comment (xhaggi) [inline xbmc/guilib/GUIEditControl.cpp]: should we move this before the call to super?

Comment (mkortstiege) [inline xbmc/guilib/GUIEditControl.cpp]: IMO it should be called last.

Comment (xhaggi) [inline xbmc/guilib/GUIEditControl.cpp]: what if `CGUIControl::SetFocus()` checks if control is invalid?

Comment (mkortstiege) [inline xbmc/guilib/GUIEditControl.cpp]: `SetFocus()` itself is not performing the invalidation and should not care about it. It's just doing the focus toggle. Mind elaborating why you think it should be called before?

Comment (xhaggi) [inline xbmc/guilib/GUIEditControl.cpp]: as discussed at IRC leave it as it is ;)

**Comment by mkortstiege** [comment]
jenkins build and merge.

**Comment by da-anda** [comment]
shouldn't the label be I*** instead of Helix? Only the backport PR should be labeled Helix, or am I wrong on that?

**Pull request merged**